runtime/kusama/src/xcm_config.rs

```diff
@@ -103,11 +109,30 @@ parameter_types! {
 	pub const MaxInstructions: u32 = 100;
 }
 
+pub struct ExponentialKsmPrice;
+impl PriceForParachainDelivery for ExponentialKsmPrice {
+	fn price_for_parachain_delivery(para: ParaId, msg: &Xcm<()>) -> MultiAssets {
+		let base = CENTS.saturating_mul(3);
+		let msg_fee = (msg.len() as u128).saturating_mul(TransactionByteFee::get());
```

Comment:
Is this `len` the number of XCM instructions or the encoded length?  
I think the encoded length would be closer to the truth since XCMs are not MEL bound.  
Or some kind of recursive counting of how many actual instructions are in there? Otherwise someone could get around it through recursion.

Reply:
Oh crud, you're right, this is the number of XCM instructions, not the encoded length of the XCM itself.